src/registry/live_check.rs

```diff
@@ -1,112 +1,340 @@
+    /// versus processing the data for every sample.
+    #[arg(long)]
+    advice_preprocessor: Option<PathBuf>,
 }
 
-/// Check the conformance level of an OTLP stream against a semantic convention registry.
-///
-/// This command starts an OTLP listener and compares each received OTLP message with the
-/// registry provided as a parameter. When the command is stopped (see stop conditions),
-/// a conformance/coverage report is generated. The purpose of this command is to be used
-/// in a CI/CD pipeline to validate the telemetry stream from an application or service
-/// against a registry.
-///
-/// The currently supported stop conditions are: CTRL+C (SIGINT), SIGHUP, the HTTP /stop
-/// endpoint, and a maximum duration of no OTLP message reception.
+/// Perform a live check on sample data by comparing it to a semantic convention registry.
 pub(crate) fn command(
-    logger: impl Logger + Sync + Clone,
-    args: &CheckRegistryArgs,
+    logger: impl Logger + Sync + Clone + 'static,
+    args: &RegistryLiveCheckArgs,
 ) -> Result<ExitDirectives, DiagnosticMessages> {
+    let mut exit_code = 0;
+    let mut output = PathBuf::from("output");
+    let output_directive = if let Some(path_buf) = &args.output {
+        output = path_buf.clone();
+        OutputDirective::File
+    } else {
+        logger.mute();
+        OutputDirective::Stdout
+    };
+
+    logger.log("Weaver Registry Live Check");
+
+    // Prepare the registry
+    logger.loading(&format!("Resolving registry `{}`", args.registry.registry));
+
     let mut diag_msgs = DiagnosticMessages::empty();
-    let policy = PolicyArgs::skip();
-    let otlp_requests = listen_otlp_requests(
-        args.otlp_grpc_address.as_str(),
-        args.otlp_grpc_port,
-        args.admin_port,
-        Duration::from_secs(args.inactivity_timeout),
-        logger.clone(),
-    )?;
 
-    // @ToDo Use the following resolved registry to check the level of compliance of the incoming OTLP messages
-    let (_resolved_registry, _) =
-        prepare_main_registry(&args.registry, &policy, logger.clone(), &mut diag_msgs)?;
+    let (registry, _) =
+        prepare_main_registry(&args.registry, &args.policy, logger.clone(), &mut diag_msgs)?;
 
     logger.loading(&format!(
-        "Checking OTLP traffic on port {}.",
-        args.otlp_grpc_port
+        "Performing live check with registry `{}`",
+        args.registry.registry
     ));
 
-    // @ToDo Implement the checking logic
-    for otlp_request in otlp_requests {
-        match otlp_request {
-            OtlpRequest::Logs(_logs) => {
-                // ToDo Implement the checking logic for logs
-                println!("Logs Request received");
-            }
-            OtlpRequest::Metrics(_metrics) => {
-                // ToDo Implement the checking logic for metrics
-                println!("Metrics Request received");
+    // Create the live checker with advisors
+    let advisors: Vec<Box<dyn Advisor>> = vec![
```

Comment:
Nit - Can we move this into a loose fucntion, like `fn default_advisors() -> Vec<Box<dyn Advisor>>`